### PR TITLE
compiles under arch

### PR DESCRIPTION
**Pull request by srtlg**
Hello, I've tried to package the classdesc-python branch for Arch Linux. The following changes seem to me reasonable for including into upstream.

- There are again conflicts with references to `std::future` like #130 
- in `scidavis/scidavis.pro` I do not get `-lpython...` without `python-config --libs --embed`

### CODE REVIEW

File: libscidavis/src/ApplicationWindow.cpp [hunk 1]
```diff
@@ -14042,7 +14042,7 @@ void ApplicationWindow::showHistory()
 
 QStringList ApplicationWindow::tableWindows()
 {
-	QList<AbstractAspect *> tables = d_project->descendantsThatInherit("future::Table");
+	QList<AbstractAspect *> tables = d_project->descendantsThatInherit("::future::Table");
```
Comment (highperformancecoder) [inline libscidavis/src/ApplicationWindow.cpp]: Is this change necessary?

File: scidavis/scidavis.pro [hunk 2]
```diff
@@ -17,7 +17,7 @@ python {
   }
   unix: {
         LIBS+=$$system($$PYTHONBIN findBoostPythonLib.py)
-        LIBS+=$$system($$PYTHONBIN-config --ldflags)
+        LIBS+=$$system($$PYTHONBIN-config --libs --embed)
```
Comment (highperformancecoder) [inline scidavis/scidavis.pro]: --embed doesn't work on my system. It causes an error. And --ldflags also includes the flags -Xlinker -export-dynamic. I can't find documentation on what is supposed to be the difference between --libs and --ldflags - do you know?

**Comment by srtlg** [comment]
On Fri, Mar 06, 2020 at 14:19 -0800, Russell Standish wrote:
> > @@ -17,7 +17,7 @@ python {
>    }
>    unix: {
>          LIBS+=$$system($$PYTHONBIN findBoostPythonLib.py)
> -        LIBS+=$$system($$PYTHONBIN-config --ldflags)
> +        LIBS+=$$system($$PYTHONBIN-config --libs --embed)
>
> --embed doesn't work on my system. It causes an error. And --ldflags also includes the flags -Xlinker -export-dynamic. I can't find documentation on what is supposed to be the difference between --libs and --ldflags - do you know?
Sorry, I don't know much about python-config. I just deduced from the error messages (linking scidavis produced lots of unresolved symbols that were all in pythonX.Y.so)

$ python-config --ldflags
 -L/usr/lib  -lcrypt -lpthread -ldl  -lutil -lm -lm
$ python-config --libs
 -lcrypt -lpthread -ldl  -lutil -lm -lm
$ python-config --embed
$ python-config --libs --embed
-lpython3.8 -lcrypt -lpthread -ldl  -lutil -lm -lm

**Comment by srtlg** [comment]
I've just checked. That is a Python 3.8 thing.

https://docs.python.org/3/whatsnew/3.8.html

"To embed Python into an application, a new --embed option must be passed to
python3-config --libs --embed to get -lpython3.8 (link the application to
libpython). To support both 3.8 and older, try python3-config --libs --embed
first and fallback to python3-config --libs (without --embed) if the previous
command fails."

They altered Python*/Misc/python-config.sh.in between 3.7 and 3.8 to have this effect
From 3.8 onwards only --embed will add -lpythonX.Y to the flags

File: scidavis/scidavis.pro (hunk 2 above)
Comment (srtlg) [reply scidavis/scidavis.pro]: What about this patch? It works under Fedora 30 (Python 3.7) and Arch (Python 3.8)

```patch
+diff -u10 -r scidavis-2.0.10-original/scidavis/scidavis.pro scidavis-2.0.10/scidavis/scidavis.pro
+--- scidavis-2.0.10-original/scidavis/scidavis.pro     2020-02-17 07:54:00.000000000 +0100
++++ scidavis-2.0.10/scidavis/scidavis.pro      2020-03-07 11:27:31.630929748 +0100
+@@ -10,21 +10,25 @@

  include(../config.pri)
  include( basic.pri )
  python {
    PYTHONBIN = $$(PYTHON)
    isEmpty( PYTHONBIN ) {
      PYTHONBIN = python3
    }
    unix: {
          LIBS+=$$system($$PYTHONBIN findBoostPythonLib.py)
 -        LIBS+=$$system($$PYTHONBIN-config --ldflags)
-+        LIBS+=$$system($$PYTHONBIN-config --libs --embed)
++        system($$PYTHONBIN-config --embed >/dev/null) {
++            LIBS+=$$system($$PYTHONBIN-config --libs --embed)
++        } else {
++            LIBS+=$$system($$PYTHONBIN-config --libs)
++        }
        }
    message($$LIBS)

    compiledScidavisrc.target = scidavisrc.pyc
    compiledScidavisrc.depends = scidavisrc.py
    compiledScidavisrc.commands = $$PYTHONBIN -m py_compile scidavisrc.py
    compiledScidavisUtil.target = scidavisUtil.pyc
    compiledScidavisUtil.depends = scidavisUtil.py
    compiledScidavisUtil.commands = $$PYTHONBIN -m py_compile scidavisUtil.py
    QMAKE_EXTRA_TARGETS += compiledScidavisrc compiledScidavisUtil
```

File: libscidavis/src/ApplicationWindow.cpp (hunk 1 above)
Comment (srtlg) [reply libscidavis/src/ApplicationWindow.cpp]: you are right. It has to remain `descendantsThatInherit("future::Table");` 
I tested it the following way
```
d_project->descendantsThatInherit("::future::Table").length() is always zero
d_project->descendantsThatInherit("future::Table").length() gives the correct number of Tables
```

File: scidavis/scidavis.pro (hunk 2 above)
Comment (highperformancecoder) [reply scidavis/scidavis.pro]: That looks like it will work. Looks like there was a change here between Python 3.7 and 3.8.

File: libscidavis/src/ApplicationWindow.cpp (hunk 1 above)
Comment (highperformancecoder) [reply libscidavis/src/ApplicationWindow.cpp]: If you remove the change on this line, we're good to go on this PR.
